### PR TITLE
chore: fix some typos in comments

### PR DESCRIPTION
**Pull request by StevenMia**

### CODE REVIEW

File: core/src/avm2/scope.rs [hunk 1]
```diff
@@ -89,11 +89,11 @@ impl<'gc> ScopeContainer<'gc> {
 /// initial creation.
 ///
 /// A ScopeChain is either created by chaining new scopes on top of an already existing
-/// ScopeChain, or if we havn't created one yet (like during script initialization), you can
+/// ScopeChain, or if we haven't created one yet (like during script initialization), you can
 /// create an empty ScopeChain with only a Domain. A ScopeChain should **always** have a Domain.
 ///
 /// ScopeChain's are copy-on-write, meaning when we chain new scopes on top of a ScopeChain, we
-/// actually create a completely brand new ScopeChain. The Domain of the ScopeChain we are chaining
```
Comment (lvyitian) [inline core/src/avm2/scope.rs:96]: The original is correct, the "brand new" is an adjective phrase.

**Comment by torokati44** [comment]
That has been done before already:
https://github.com/ruffle-rs/ruffle/pull/14803
https://github.com/ruffle-rs/ruffle/pull/15303

But it shows a lot of false positives, a lot of typos that we intend to keep, and a few that are in "imported" tests, which is not necessarily worth fixing.

File: core/src/avm2/e4x.rs [hunk 2]
```diff
@@ -1343,7 +1343,7 @@ pub fn to_xml_string<'gc>(
             .get_public_property("prettyIndent", activation)
             .expect("prettyIndent should be set")
             .coerce_to_i32(activation)
-            .expect("shouldnt error");
+            .expect("shouldn't error");
```
Comment (torokati44) [inline core/src/avm2/e4x.rs:1346]: This one is not in a comment, but oh well... :)

**Review by torokati44 — APPROVED** [review]
Thank you!

**Pull request merged**